File: shared/src/commonMain/kotlin/com/siptech/siptrack/models/Drink.kt
package com.siptech.siptrack.models

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class Drink(
    val id: String,
    val sessionId: String,
    val name: String,
    val category: DrinkCategory,
    val abvPercent: Float,
    val volumeOz: Float,
    val loggedAt: Instant,
    val productId: String? = null,
    val notes: String? = null,
    val isMicroDose: Boolean = false,
) {
    /** Standard drinks = (volumeOz × ABV × 0.816) / 0.6 */
    val standardDrinks: Float
        get() = (volumeOz * (abvPercent / 100f) * 0.816f) / 0.6f

    /** Alcohol grams for Widmark formula */
    val alcoholGrams: Float
        get() = volumeOz * 29.5735f * (abvPercent / 100f) * 0.789f

    /** Estimated calories (7 cal/gram of alcohol) */
    val calories: Float
        get() = alcoholGrams * 7f
}

@Serializable
enum class DrinkCategory(val displayName: String, val defaultAbv: Float, val defaultVolumeOz: Float) {
    BEER("Beer", 5.0f, 12.0f),
    WINE("Wine", 12.0f, 5.0f),
    SPIRIT("Spirit", 40.0f, 1.5f),
    COCKTAIL("Cocktail", 15.0f, 4.0f),
    HARD_SELTZER("Hard Seltzer", 5.0f, 12.0f),
    CIDER("Cider", 5.5f, 12.0f),
    MICRO_TASTE("Taste", 40.0f, 0.5f),
    CUSTOM("Custom", 5.0f, 12.0f)
}

File: SipTrack/Models/Drink.cs
using SQLite;
using System;

namespace SipTrack.Models
{
    public enum DrinkCategory
    {
        Beer,
        Wine,
        Spirit,
        Cocktail,
        Custom
    }

    [Table("Drinks")]
    public class Drink
    {
        [PrimaryKey, AutoIncrement]
        public int Id { get; set; }

        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        public DrinkCategory Category { get; set; } = DrinkCategory.Custom;

        /// <summary>Alcohol by volume as percentage (e.g. 5.0 for 5%)</summary>
        public double AbvPercent { get; set; }

        /// <summary>Volume in fluid ounces</summary>
        public double VolumeOz { get; set; }

        /// <summary>Volume in millilitres (kept in sync with VolumeOz)</summary>
        public double VolumeMl
        {
            get => VolumeOz * 29.5735;
            set => VolumeOz = value / 29.5735;
        }

        /// <summary>
        /// Standard drinks = (VolumeOz * AbvPercent/100 * 0.816) / 0.6
        /// (based on 14g pure alcohol = 1 standard drink)
        /// </summary>
        public double StandardDrinks => (VolumeOz * (AbvPercent / 100.0) * 0.816) / 0.6;

        /// <summary>Estimated calories: 7 kcal per gram of alcohol</summary>
        public double CaloriesEstimate
        {
            get
            {
                double alcoholGrams = VolumeOz * 29.5735 * (AbvPercent / 100.0) * 0.789;
                return alcoholGrams * 7.0;
            }
        }

        public DateTime LoggedAt { get; set; } = DateTime.UtcNow;

        public int SessionId { get; set; }

        public bool IsCustom { get; set; }

        [MaxLength(500)]
        public string Notes { get; set; } = string.Empty;

        // ── Ignored (not stored in DB) ───────────────────────────────────
        [Ignore]
        public string CategoryIcon => Category switch
        {
            DrinkCategory.Beer => "🍺",
            DrinkCategory.Wine => "🍷",
            DrinkCategory.Spirit => "🥃",
            DrinkCategory.Cocktail => "🍹",
            _ => "🥤"
        };

        [Ignore]
        public string DisplayVolume => $"{VolumeOz:0.#} oz";

        [Ignore]
        public string DisplayAbv => $"{AbvPercent:0.#}%";
    }
}
